lib/utils.nix

```
@@ -172,6 +172,41 @@ rec {
     in
     lib.literalExpression exp;
 
+  /**
+    Convert one or several conditional warnings to a final warning list.
+
+    # Example
+
+    ```nix
+    warnings = mkWarnings [
+      {
+        condition = plugins.foo.settings.barIntegration && (!plugins.bar.enable);
+        message = "`barIntegration` is enabled but the `bar` plugin is not."
+      }
+    ];
+    ```
+
+    # Type
+    ```
+    mkWarnings :: String -> List -> List
+    ```
+  */
+  mkWarnings =
+    scope: w:
+    let
+      warnings = if !lib.isList w then [ w ] else w;
+
+      processWarning =
+        warning:
+        lib.optional (warning.condition or true) (
+          let
+            message = if lib.isString warning then warning else warning.message;
+          in
+          "Nixvim (${scope}): ${message}"
+        );
+    in
+    lib.flatten (map processWarning warnings);
```

```suggestion
    builtins.concatMap processWarning (lib.toList w);
```

Lets not use a seldgehammer (`flatten`) to drive in a nail (list of lists).

Also, I think we can inline `warnings` here without loosing any readability.